packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -279,6 +283,11 @@ export class ImageEditPlugin implements ImageEditor, EditorPlugin {
         }
     }
 
+    private beforeSetContentHandler(editor: IEditor, newContent: string) {
+        newContent.replace(IMAGE_EDIT_DATASET, '');
```

Comment:
are you sure this can work since you didn't use the return value?

Reply:
Instead of changing the source HTML, I prefer to do it in DOM tree, when handling ContentChangedEvent and check change source is setContent